internal/hydra/client.go

```diff
@@ -19,6 +20,13 @@ func (c *Client) MetadataApi() client.MetadataApi {
 	return c.c.MetadataApi
 }
 
+// We implement the device API logic, as the upstream sdk does not support it.
+// Otherwise we would have to fork the sdk
+// TODO(nsklikas): Remove this once upstream hydra supports the device flow
+func (c *Client) DeviceApi() DeviceApi {
+	return c.deviceApi
+}
```

Comment:
had another thought, would we be able to embed these into the oauth2Api instead?

```
 func (c *Client) OAuth2Api() OAuth2Api {
.
.
.

type OAuth2Api interface {
   client.OAuth2Api
   AcceptUserCodeRequest(context.Context) ApiAcceptUserCodeRequestRequest
   AcceptUserCodeRequestExecute(ApiAcceptUserCodeRequestRequest) (*hClient.OAuth2RedirectTo, *http.Response, error)
}
``` 

we would wrap only the oauth2 client basically